File: frontend/src/types/recipe.ts
export interface Receta {
  nombre: string
  busqueda_imagen?: string
  emoji: string
  descripcion: string
  tiempo_minutos: number
  porciones: number
  dificultad: 'fácil' | 'intermedio' | 'difícil'
  calorias_aprox: number
  ingredientes_usados: string[]
  ingredientes_extra: string[]
  pasos: string[]
  tip_chef: string
}

export interface RecetasResponse {
  recetas: Receta[]
}

export interface Filtros {
  dieta: 'ninguna' | 'vegetariano' | 'vegano' | 'sin_gluten' | 'sin_lactosa'
  porciones: number
  complejidad: 'cualquiera' | 'fácil' | 'intermedio' | 'difícil'
  tiempo_max: number
  momento: 'cualquiera' | 'desayuno' | 'almuerzo' | 'cena'
  sabor: 'cualquiera' | 'dulce' | 'salado'
}

export const DEFAULT_FILTROS: Filtros = {
  dieta: 'ninguna',
  porciones: 2,
  complejidad: 'cualquiera',
  tiempo_max: 60,
  momento: 'cualquiera',
  sabor: 'cualquiera',
}

export interface RecetaCompartida {
  id?: string
  receta: Receta
  timestamp: string
  autor_nombre?: string
}

export const SUGERENCIAS = [
  'tomate', 'cebolla', 'ajo', 'huevo', 'pollo', 'papa',
  'arroz', 'pasta', 'zanahoria', 'queso', 'atún', 'limón',
  'espinaca', 'pimiento', 'hongos', 'zapallo', 'manteca', 'crema',
]

File: backend/schemas/recipe.py
import uuid
from typing import Any

from pydantic import BaseModel, ConfigDict, Field


class Filtros(BaseModel):
    dieta: str = "ninguna"
    porciones: int = 2
    complejidad: str = "cualquiera"
    tiempo_max: int = 60
    momento: str = "cualquiera"
    sabor: str = "cualquiera"


class RecetaRequest(BaseModel):
    ingredientes: list[str]
    filtros: Filtros = Field(default_factory=Filtros)


class Receta(BaseModel):
    nombre: str
    busqueda_imagen: str = ""
    emoji: str
    descripcion: str
    tiempo_minutos: int
    porciones: int
    dificultad: str
    calorias_aprox: int
    ingredientes_usados: list[str]
    ingredientes_extra: list[str]
    pasos: list[str]
    tip_chef: str


class RecipeCreate(BaseModel):
    title: str = Field(min_length=1, max_length=255)
    content: dict[str, Any]


class RecipeRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    user_id: uuid.UUID
    title: str
    content: dict[str, Any]
